src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -90,9 +90,14 @@ object RefChecks {
         if (pclazz.is(Sealed) && pclazz.associatedFile != cls.associatedFile)
           ctx.error(d"cannot extend sealed $pclazz in different compilation unit", cls.pos)
         checkSelfConforms(parent, "illegal inheritance", "parent")
+
+        // Maintain children table for sealed class to facilitate exhaustivity
+        // and redundancy check of patttern matching
+        if (pclazz.is(Sealed)) pclazz.addChild(cls)
       }
```

Comment:
@odersky: Do you think the list of children of a sealed class should be stored in the `ClassSymbol` of the class?